update/sync/syncAccountsDBs.go

```diff
@@ -0,0 +1,210 @@
+func NewSyncAccountsDBsHandler(args ArgsNewSyncAccountsDBsHandler) (*syncAccountsDBs, error) {
+	if check.IfNil(args.AccountsDBsSyncers) {
+		return nil, update.ErrNilAccountsDBSyncContainer
+	}
+
+	st := &syncAccountsDBs{
+		tries:              newConcurrentTriesMap(),
+		accountsBDsSyncers: args.AccountsDBsSyncers,
+		activeAccountsDBs:  make(map[state.AccountsDbIdentifier]state.AccountsAdapter),
+		synced:             false,
+		mutSynced:          sync.Mutex{},
+	}
+	for key, value := range args.ActiveAccountsDBs {
+		st.activeAccountsDBs[key] = value
+	}
+
+	return st, nil
+}
+
+// SyncAccountsDBsFrom syncs all the state tries from an epoch start metachain
+func (st *syncAccountsDBs) SyncTriesFrom(meta *block.MetaBlock, waitTime time.Duration) error {
+	if !meta.IsStartOfEpochBlock() {
+		return update.ErrNotEpochStartBlock
+	}
+
+	var errFound error
+	mutErr := sync.Mutex{}
+
+	st.synced = false
+	wg := sync.WaitGroup{}
+	wg.Add(1 + len(meta.EpochStart.LastFinalizedHeaders))
+
+	chDone := make(chan bool)
+	go func() {
+		wg.Wait()
+		chDone <- true
+	}()
+
+	go func() {
+		errMeta := st.syncMeta(meta)
+		if errMeta != nil {
+			mutErr.Lock()
+			errFound = errMeta
+			mutErr.Unlock()
+		}
+		wg.Done()
+	}()
+
+	for _, shData := range meta.EpochStart.LastFinalizedHeaders {
+		go func(shardData block.EpochStartShardData) {
+			err := st.syncShard(shardData)
+			if err != nil {
+				mutErr.Lock()
+				errFound = err
+				mutErr.Unlock()
+			}
+			wg.Done()
+		}(shData)
+	}
+
+	err := WaitFor(chDone, waitTime)
+	if err != nil {
+		return err
+	}
+
+	if errFound != nil {
+		return errFound
+	}
+
+	st.mutSynced.Lock()
+	st.synced = true
+	st.mutSynced.Unlock()
+
+	return nil
+}
+
+func (st *syncAccountsDBs) syncMeta(meta *block.MetaBlock) error {
+	err := st.syncAccountsOfType(state.UserAccount, state.UserAccountsState, core.MetachainShardId, meta.RootHash)
+	if err != nil {
+		return err
+	}
+
+	err = st.syncAccountsOfType(state.ValidatorAccount, state.PeerAccountsState, core.MetachainShardId, meta.ValidatorStatsRootHash)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (st *syncAccountsDBs) syncShard(shardData block.EpochStartShardData) error {
+	err := st.syncAccountsOfType(state.UserAccount, state.UserAccountsState, shardData.ShardId, shardData.RootHash)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (st *syncAccountsDBs) syncAccountsOfType(accountType state.Type, trieID state.AccountsDbIdentifier, shardId uint32, rootHash []byte) error {
+	accAdapterIdentifier := genesis.CreateTrieIdentifier(shardId, accountType)
+
+	success := st.tryRecreateTrie(shardId, accAdapterIdentifier, trieID, rootHash)
+	if success {
+		return nil
+	}
+
+	accountsDBSyncer, err := st.accountsBDsSyncers.Get(accAdapterIdentifier)
+	if err != nil {
+		return err
+	}
+
+	err = accountsDBSyncer.SyncAccounts(rootHash)
+	if err != nil {
+		// critical error - should not happen - maybe recreate trie syncer here
```

Comment:
This might and will happen if the sync stops as the network will be down longer than hardcoded wait time. Might have a retry loop here.

Reply:
added TODO for now.